From 96b30e3077b045b99a3b44dc35a4dfc9adc6877b Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 23 Jun 2019 11:50:18 +0300
Subject: [PATCH] [SW-0] gaudi: on reset switch uart/spi clk to ref

The BTL is initializes the baud rate of the UART to a certain value
which cannot be manipulated in run time. This means we need to setup the
UART clock to a frequency which will suit this limitation.
This frequency is 50MHz, so switch the UART/SPI clock (PCI PLL DIV
output 2) to reference clock before reset.

Change-Id: Ica79cc6e7abbeedac1af47c6fbb2bbea0e7e973e
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index 1735fc1..2a65bcc 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -8,6 +8,7 @@
 #include <stdbool.h>
 
 #include <psoc_global_conf_regs.h>
+#include <psoc_pci_pll_regs.h>
 
 #include <preboot.h>
 #include <pll.h>
@@ -330,6 +331,12 @@
 
 static void gaudi_reset_soc(void)
 {
+	/* move the UART/SPI clock to reference clock */
+	hl_pll_udelay(10);
+	regmap_writel(HL_PLL_REF_CLK, mmPSOC_PCI_PLL_DIV_SEL_2);
+	hl_pll_udelay(1);
+	hl_uart_init();
+
 	regmap_writel(0xFFE00000, mmPSOC_GLOBAL_CONF_SW_ALL_RST_CFG_L);
 	regmap_writel(0x003FFFFF, mmPSOC_GLOBAL_CONF_SW_ALL_RST_CFG_H);
 	regmap_writel(1, mmPSOC_GLOBAL_CONF_SW_ALL_RST);
